pkg/secrets/secrets.go

```diff
@@ -1,57 +1,71 @@
 package secrets
 
 import (
+	"encoding/json"
 	"io/ioutil"
 	"os"
 	"os/exec"
-	"path/filepath"
 )
 
-const SecretsFileName = "secrets.oya"
+func Decrypt(path string) ([]byte, bool, error) {
+	if ok, err := isSopsFile(path); !ok || err != nil {
+		return nil, false, err
+	}
 
-func Decrypt(workDir string) ([]byte, error) {
 	var output []byte
-	file := filepath.Join(workDir, SecretsFileName)
-	if _, err := os.Stat(file); err != nil {
-		return output, ErrNoSecretsFile{FileName: SecretsFileName, OyafilePath: workDir}
+	if _, err := os.Stat(path); err != nil {
+		return output, false, ErrNoSecretsFile{Path: path}
 	}
-	decryptCmd := exec.Command("sops", "-d", file)
+	decryptCmd := exec.Command("sops", "-d", path)
 	decrypted, err := decryptCmd.CombinedOutput()
 	if err != nil {
-		return output, ErrSecretsFailure{FileName: SecretsFileName, OyafilePath: workDir, CmdError: string(decrypted)}
+		return output, false,
+			ErrSecretsFailure{Path: path, CmdError: string(decrypted)}
 	}
-	return decrypted, nil
+	return decrypted, true, nil
 }
 
-func Encrypt(workDir string) error {
-	file := filepath.Join(workDir, SecretsFileName)
-	if alreadyEncrypted(file) {
-		return ErrSecretsAlreadyEncrypted{FileName: SecretsFileName, OyafilePath: workDir}
+func Encrypt(path string) error {
+	if alreadyEncrypted(path) {
+		return ErrSecretsAlreadyEncrypted{Path: path}
 	}
-	cmd := exec.Command("sops", "-e", file)
+	cmd := exec.Command("sops", "-e", path)
 	encoded, err := cmd.CombinedOutput()
 	if err != nil {
-		return ErrSecretsFailure{FileName: SecretsFileName, OyafilePath: workDir, CmdError: string(encoded)}
+		return ErrSecretsFailure{Path: path, CmdError: string(encoded)}
 	}
-	fi, err := os.Stat(file)
+	fi, err := os.Stat(path)
 	if err != nil {
 		return err
 	}
-	err = ioutil.WriteFile(file, encoded, fi.Mode())
+	err = ioutil.WriteFile(path, encoded, fi.Mode())
```

Comment:
Hm, just a note: don't like overwrite in place without option to avoid it ;) Maybe this function could take `src path, dst path` and for now caller would pass same as both args?